app/src/main/java/com/zulip/android/activities/MessageListFragment.java

```diff
@@ -206,11 +207,63 @@ public boolean onContextItemSelected(MenuItem item) {
                 copyMessage(message);
                 Toast.makeText(getContext(), R.string.message_copied, Toast.LENGTH_SHORT).show();
                 return true;
+            case R.id.star_message:
+                starMessage(message);
+                return true;
+            case R.id.un_star_message:
+                unStarMessage(message);
+                return true;
             default:
                 return super.onContextItemSelected(item);
         }
     }
 
+    /**
+     * Stars a message passed as parameter
+     *
+     * @param message Message to be starred
+     */
+    private void starMessage(final Message message) {
+        AsyncStar asyncStar = new AsyncStar(((ZulipActivity) getActivity()), message, "add");
+        asyncStar.setCallback(new ZulipAsyncPushTask.AsyncTaskCompleteListener() {
+            @Override
+            public void onTaskComplete(String result, JSONObject jsonObject) {
+                message.setMessageStar(true);
```

Comment:
here I think it makes sense to move the UI changing code before updating the server. We might want to show user the message is starred immediately and then in case of failure unstar it and show a toast message.